test/libsolidity/ASTJSONTest.cpp

```diff
@@ -105,6 +116,15 @@ ASTJSONTest::ASTJSONTest(string const& _filename)
 			);
 			source = string();
 		}
+		else if (boost::algorithm::starts_with(line, failMarker))
+		{
+			string state = line.substr(failMarker.size(), line.size());
```

Comment:
The second argument is the length:
```suggestion
			string state = line.substr(failMarker.size(), line.size() - failMarker.size());
```
I mean, your code works because if you specify a value that exceeds string length it will be truncated. But it still looks a bit misleading. If that was your intention, I'd just leave the second arg at the default:

```suggestion
			string state = line.substr(failMarker.size());
```